engine/src/main/java/org/terasology/world/chunks/internal/ChunkImpl.java

```diff
@@ -64,6 +65,9 @@
     private static final DecimalFormat PERCENT_FORMAT = new DecimalFormat("0.##");
     private static final DecimalFormat SIZE_FORMAT = new DecimalFormat("#,###");
 
+    private final int verticalChunkMeshSegments = CoreRegistry.get(Config.class).getSystem().getVerticalChunkMeshSegments();
+    private final int chunkHalfHeight = ChunkConstants.SIZE_Y / verticalChunkMeshSegments / 2;
```

Comment:
To use a simple example of a chunk which is two blocks high, and ignoring the bottom and top ends of the chunk:
With VERTICAL_SEGMENTS = 1, you end up with a single mesh, containing 8 faces, 16 triangles and 32 vertices - all the faces of two blocks.
With VERTICAL_SEGMENTS = 2, you end up with two mesh, containing 4 faces, 8 triangles and 16 vertices each. One mesh contains the bottom block, the other the top block.

Faces are not split by VERTICAL_SEGMENTS, the chunk itself is partitioned and a mesh generated for each partition.

And yes, this can be handled by another PR.